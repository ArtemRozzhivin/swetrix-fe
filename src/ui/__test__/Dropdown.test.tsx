/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Dropdown from '../Dropdown'

describe('Dropdown', () => {
  const items = ['Item 1', 'Item 2', 'Item 3']
  const onSelect = jest.fn()
  const defaultProps = {
    title: 'Dropdown Title',
    onSelect,
    items,
  }

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('should render without errors', () => {
    const { getByText } = render(<Dropdown {...defaultProps} />)
    expect(getByText(defaultProps.title)).toBeInTheDocument()
  })

  it('should render the description when provided', () => {
    const description = 'Dropdown description'
    const { getByText } = render(<Dropdown {...defaultProps} desc={description} />)
    expect(getByText(description)).toBeInTheDocument()
  })

  it('should call onSelect when an item is clicked', () => {
    const { getByText } = render(<Dropdown {...defaultProps} />)
    const btnOpen = getByText(defaultProps.title)
    fireEvent.click(btnOpen)
    const item = getByText(items[0])
    fireEvent.click(item)
    expect(onSelect).toHaveBeenCalledWith(items[0])
  })
})
