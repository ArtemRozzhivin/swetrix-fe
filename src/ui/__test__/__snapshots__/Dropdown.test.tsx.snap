// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snapshot 1`] = `
<div>
  <div
    class="relative inline-block text-left"
    data-headlessui-state=""
  >
    <div>
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="inline-flex w-full rounded-md border border-gray-300 shadow-sm px-3 md:px-4 py-2 bg-white text-sm font-medium text-gray-700 dark:text-gray-50 dark:border-gray-800 dark:bg-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-indigo-500 justify-center"
        data-headlessui-state=""
        id="headlessui-menu-button-:r7:"
        type="button"
      >
        Dropdown Title
        <svg
          aria-hidden="true"
          class="-mr-1 ml-2 h-5 w-5"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
